JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/query/RemoteRDBStoreClient.java

```diff
@@ -0,0 +1,114 @@
+package uk.ac.cam.cares.jps.base.query;
+
+import uk.ac.cam.cares.jps.base.exception.JPSRuntimeException;
+import java.sql.*;
+
+/**
+ * This class allows to establish connection with relational databases (RDBs)
+ * to perform query and update operations.
+ *
+ * It is also used to obtain the connection object after the connection to the RDB
+ * has been established.
+ *
+ * It requires to set the URL and credentials for the database.
+ * Example URL: jdbc:postgresql://host.docker.internal:5432/timeseries
+ *
+ * @author Mehal Agarwal (ma988@cam.ac.uk)
+ *
+ */
+
+public class RemoteRDBStoreClient {
+
+    //RDB Connection object
+    private Connection conn;
+    Statement stmt;
+    // URL and credentials for the relational database
+    private String rdbURL;
+    private String rdbUser;
+    private String rdbPassword;
+
+    /**
+     * A constructor defined to initialise the URL, username and password to connect to the RDB
+     * @param rdbURL
+     * @param user
+     * @param password
+     */
+    public RemoteRDBStoreClient(String rdbURL, String user, String password) {
+        this.rdbURL = rdbURL;
+        this.rdbUser = user;
+        this.rdbPassword = password;
+    }
+
+    /**
+     * Establish connection to RDB and set Statement object
+     * @return connection object to the RDB
+     */
+    public Connection getConnection(){
+        try {
+            Class.forName("org.postgresql.Driver");
+            conn = DriverManager.getConnection(this.rdbURL, this.rdbUser, this.rdbPassword);
+            stmt = conn.createStatement(java.sql.ResultSet.TYPE_FORWARD_ONLY, java.sql.ResultSet.CONCUR_READ_ONLY);
+        } catch (SQLException | ClassNotFoundException e) {
+            throw new JPSRuntimeException(e.getMessage(), e);
+        }
+        return conn;
+    }
```

Comment:
A better practice here is to catch the exceptions separately and give a different message, e.g.
` catch (ClassNotFoundException e) {
    throw new JPSRuntimeException ("Failed to load postgresql driver",e)"`
    
The information in e.getMessage() is already in "e", so you're probably duplicating information here

Reply:
sure, will make this change.
Would if make sense to replace it with this catch block:
`catch (Exception e) {
                System.out.println("Connecting failed: " + this.rdbURL);
		throw new JPSRuntimeException(exceptionPrefix + "Establishing database connection failed");
}`
or to handle each exception seperately?

Reply:
where exceptionPrefix = this.getClass().getSimpleName() + ": "

Reply:
It's better to handle each separately, SQLException for connection failure, ClassNotFoundException for Failing to load class